ibis/expr/types.py

```diff
@@ -1039,6 +1039,54 @@ def type(self):
         return dt.int64
 
 
+class UInt8Value(IntegerValue):
+
+    _implicit_casts = set([
+        'uint8', 'uint16', 'uint32', 'uint64',
+        'float', 'double', 'decimal'
+    ])
+
+    def type(self):
+        return dt.uint8
+
+
+class UInt16Value(IntegerValue):
+
+    _implicit_casts = set(['uint32', 'uint64',
+                           'float', 'double', 'decimal'])
+
+    def type(self):
+        return dt.uint16
+
+
+class UInt32Value(IntegerValue):
+
+    _implicit_casts = set(['uint64', 'float', 'double', 'decimal'])
+
+    def type(self):
+        return dt.uint32
+
+
+class UInt64Value(IntegerValue):
+
+    _implicit_casts = set(['float', 'double', 'decimal'])
+
+    def type(self):
+        return dt.uint64
+
+
+class HalffloatingValue(NumericValue):
```

Comment:
There's no need for this class. Just have `HalffloatValue` subclass directly from `NumericValue`.

Reply:
done

Reply:
I don't think this has been addressed. I just checked out your branch and this is still there.

Reply:
fixed